File: ColorChoice/ColorChoice.java
public class ColorChoice{
    
  public static long chooseColor(long m, int n){
    long fac = 1;

    for(int i = 1 ;i <= n /2 ; i++){
        fac = fac * (n + 1 - i) / i; 
        if(fac == m)
            return i;
    }

    return -1;
  }

  public static void main(String[] args){
      System.out.println(ColorChoice.chooseColor(221,12));
  }
}

File: ColorChoice/ColorChoice.py
factorial = []
def generateList(n):
    factorial.append(1)
    factorial.append(1)
    for i in range(2,n+1):
        factorial.append(i * factorial[i-1])
def calculateFac(n,i):
    return int((factorial[n]) / (factorial[i] * factorial[n-i]))
def ColorChoice(m,n):
    for i in range(1,int(n/2)+1):
        if m == calculateFac(n,i):
            return i
    return -1

def checkChoose(m,n):
    generateList(n)
    return ColorChoice(m,n)

print (checkChoose(221,12))


